src/main/java/de/tum/in/www1/artemis/service/linkpreview/ogparser/Content.java

```diff
@@ -0,0 +1,36 @@
+package de.tum.in.www1.artemis.service.linkpreview.ogparser;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Represents the content of an Open Graph meta tag.
+ */
+public class Content {
```

Comment:
This and maybe some of the following classes can be a record

Reply:
Hmm... which ones exactly? If i decide to convert to record i might have to refactor many places.

Reply:
I'd suggest to do this for every class that only holds values that will never change.
It's not a blocker for me...